### PR TITLE
Add APHL Exposure Notifications

### PR DESCRIPTION
**Pull request by stephherbers**
https://dimagi-dev.atlassian.net/secure/RapidBoard.jspa?rapidView=143&projectKey=USH&modal=detail&selectedIssue=USH-224&assignee=5f314baf6db35e0039014d73 

##### SUMMARY

We want to add the [APHL Exposure Notifications](https://github.com/google/exposure-notifications-verification-server/blob/main/docs/api.md) as a server option for receiving OTPs. The desired workflow mirrors what is established for the current feature flag of [enable retrieving OTPs from a GAEN Server](https://confluence.dimagi.com/display/ccinternal/COVID%3A+Enable+retrieving+OTPs+from+a+GAEN+Server)

##### FEATURE FLAG

[COVID: Enable retrieving OTPs from a GAEN Server](https://confluence.dimagi.com/display/ccinternal/COVID%3A+Enable+retrieving+OTPs+from+a+GAEN+Server)

##### RISK ASSESSMENT / QA PLAN

I am not requesting QA. I have been testing locally with the testing server mimicking the desired workflow. 

##### PRODUCT DESCRIPTION

Luckily, only small changes needed to be made to the existing feature flag to accommodate this new server. 
From the frontend, in the "GAEN OTP Server Config" settings page, a dropdown is added for "GAEN Server Type" to choose between the Nearform OTP Server (previously used) and APHL Exposure Notifications. 
Screen Shot 2020-10-20 at 9 31 35 AM
Screen Shot 2020-10-20 at 9 31 48 AM
And when the OTP code is requested, it launches a new page displaying the code like before: 
Screen Shot 2020-10-20 at 9 30 34 AM

From the backend, get_otp_response and get_post_data_for_otp needed separate cases for the two types of servers. APHL Exposure Notifications require slightly different parameters and json format for the post_data. Also, a different format for the headers. 

From the documentation for  [APHL Exposure Notifications](https://github.com/google/exposure-notifications-verification-server/blob/main/docs/api.md), the particular API we are using is api/issue. This requires an admin-level API key and for the URL to end with "api/issue." The docs will have to be updated to make this super obvious. 

The migration adds the new column containing the "GAEN Server Type" and sets the default to the previous server type, Nearform. I don't think this will cause a huge delay during so it can be merged normally, but let me know if you think otherwise.

### CODE REVIEW

**Comment by orangejenny** [comment]
@stephherbers FYI you can edit `dialer_utils.js` [like this](https://eslint.org/docs/2.13.1/user-guide/configuring#disabling-rules-with-inline-comments) to turn off stickler. Usually not good practice, but this file is boilerplate copied from some AWS tutorial so I think it's fine here.

File: corehq/apps/integration/forms.py
Comment (orangejenny) [inline corehq/apps/integration/forms.py:108]: This can reference `GaenOtpServerSettings.GAEN_SERVER_TYPE_OPTIONS`, which could be renamed to something like `SERVER_OPTIONS` since it's in the `GaenOtpServerSettings` model.

File: corehq/apps/integration/forms.py
Comment (orangejenny) [inline corehq/apps/integration/forms.py:105]: Doesn't look like this is used.

File: corehq/apps/integration/models.py
Comment (orangejenny) [inline corehq/apps/integration/models.py:17]: translations

File: corehq/apps/integration/forms.py
Comment (orangejenny) [inline corehq/apps/integration/forms.py:109]: Nitpick: "server type"?

File: corehq/apps/integration/models.py
Comment (orangejenny) [reply corehq/apps/integration/models.py:17]: I'd also make the values into constants.

File: corehq/apps/integration/views.py
Comment (orangejenny) [inline corehq/apps/integration/views.py:131]: Do you definitely need to call `json.dumps`? I'd expect `requests` to known how to handle a dict when you're also passing the content type.

File: corehq/apps/integration/static/integration/js/dialer/dialer_utils.js
Comment (orangejenny) [inline corehq/apps/integration/static/integration/js/dialer/dialer_utils.js:1]: It looks like all the js changes are whitespace, plus the eslint tag. Can you just remove them? It'll save your name from coming up when future devs blame this file.

**Review by orangejenny — COMMENTED** [review]
Looks good. One design comment: I remember we talked about directly modifying `get_post_data_for_otp ` and `get_otp_response`, but seeing the code now, I think this would be better encapsulated if that logic were moved into `GaenOtpServerSettings`, so the model itself controls the type-based behavior.

File: corehq/apps/integration/models.py
Comment (orangejenny) [reply corehq/apps/integration/models.py:17]: ...and by "values" I meant the opposite, the keys.

File: corehq/apps/integration/views.py
Comment (stephherbers) [reply corehq/apps/integration/views.py:131]: when I don't have `json.dumps`, there is a 400 error "malformed json at position 2"

Comment (orangejenny) [reply corehq/apps/integration/views.py:131]: Okay, cool, doesn't sound worth digging into more since it works.

**Review by orangejenny — COMMENTED** [review]
Looks good, just holding off on approving pending the UX issue we chatted about offline.

**Review by orangejenny — APPROVED** [review]

**Pull request merged**
